core/src/test/java/smile/regression/GradientTreeBoostTest.java

```diff
@@ -159,7 +159,7 @@ public void testHuber() {
     @Test
     public void testShap() {
         MathEx.setSeed(19650218); // to get repeatable results.
-        GradientTreeBoost model = GradientTreeBoost.fit(BostonHousing.formula, BostonHousing.data, Loss.ls(), 100, 6, 100, 5, 0.05, 0.7);
+        GradientTreeBoost model = GradientTreeBoost.fit(BostonHousing.formula, BostonHousing.data, Loss.ls(), 100, 3, 10, 5, 0.01, 1);
```

Comment:
subsmple = 1.0 doesn't make much sense for gradient tree boosting. It is why I change it to 0.7. maxDepth = 3 is also too shallow to test some cases. Can you get the values for my settings?

Reply:
ignoring above change in `unwindPathSum`, no other obvious calculation issue by me for the implementation. the test shap number is as below for your parameter setting:
```
----- SHAP -----
ZN              0.0052
CHAS            0.0067
INDUS           0.0294
RAD             0.0316
B               0.0482
TAX             0.0807
CRIM            0.0994
AGE             0.1206
NOX             0.1341
PTRATIO         0.1530
DIS             0.1581
RM              0.5589
LSTAT           0.9871
```

Reply:
I double checked unwindPathSum and also verify the math formula. I think that my implementation is correct (also faster).

Thanks for the new SHAP values. These are much closer to my numbers. But why the numbers in the older PR are much bigger?

Reply:
> I double checked unwindPathSum and also verify the math formula. I think that my implementation is correct (also faster).
> 
> Thanks for the new SHAP values. These are much closer to my numbers. But why the numbers in the older PR are much bigger?

different GradientTreeBoost parameter make the majority of difference IMO. not the shap calculation.